packages/apollo-server-defer/README.md

```diff
@@ -0,0 +1,186 @@
+
+```
+// Initial Response
+{
+  "data": {
+    "asset": {
+      "title": "Westworld",
+    }
+  }
+}
+
+// Stream in deferred fields as patches
+{
+  "path": ["video", "progress"],
+  "data": {
+    "percent": 20,
+    "position": 20
+  }
+}
+{
+  "path": ["video", "comments"],
+  "data": [{
+    "userId": "123",
+    "text": "..."
+  }]
+}
+```
+
+### Why is @defer an ideal solution?
+
+- Query remains super easy to read 👍
+- Stays true to Apollo's declarative approach to data loading - everything is encapsulated in a single query component 😎
+- Greatly reduces boilerplate code to manage multiple requests, a huge value-add for users that use both Apollo Server AND Client together.
+
+## Implementation Details
+
+### Patch Data Format:
+
+- Proposed format for `ExecutionPatchResult`
+  ```graphql
+  {
+    path: [string]!
+    data: {}
+    errors: []
+  }
+  ```
+- We should use the same patch format for @defer/stream/live support.
+
+### Apollo Server:
+
+In order to support @defer, there are significant changes to be made to the execution phase of GraphQL.
+
+- Extending graphql-js execution to support deferred responses.
+- Defining a new response type that wraps the initial response and an asynchronous stream of patches. 
+- Maximize code reuse by exporting types and utility functions from graphql.js, making a PR if necessary.
+- Restrict peer dependency on graphql.js to versions that we have tested with.
+
+### Apollo Client:
+
+- Should contain the logic to merge patches in and update the UI as data gets streamed in. 
+- Initial implementation of @defer support should come as a Apollo Link. Reads from a socket connection or some other event stream, keeps the partial response in memory, merging patches as they come and pushing it through the link stack.
+
+### Errors:
+
+This refers specifically to the errors that occur when streaming deferred fields. Errors on the initial response will be handled as per normal.
+
+- Resolver level errors can be sent along with the patch. These will be merged with the errors array to create the full response.
+- Network errors (i.e when the websocket connection is broken). Ideally, a network error should not nullify the results that have already been sent succesfully. However, the client should expose the network status to indicate whether all the patches have been sent successfully.  
+
+### Transport:
+
+- Upgrading `HttpLink` to accept multipart http responses from `apollo-server-express`. This is the default solution, since it is lightweight compared to websockets, and requires no additional set up from the user. Note that we are using the ReadableStream API, which is available on most modern browsers. 
```

Comment:
Does this mean it will crash on IE? Or is there a fallback?

Reply:
For browsers that do not support ReadableStreams, I think the safest fallback is to strip out @defer from the query, and treat it as a normal request, and give warnings to the user. The only way I know of to get defer support in IE is to use a different transport layer like websockets?